### PR TITLE
add visitor pattern implementation for Poco::Dynamic::Var

### PR DESCRIPTION
**Pull request by bas524**
Complimentary for #2291
It's a simple implementation of runtime visitor
Main idea is using unique handler for every type which can be saved in the Dynamic::Var

As handler you can use lambda, std::function and function pointer with signature:
```cpp
std::function<void(const T&)>;
void (*)(const T &);
```

example
```cpp
// ...
Visitor visitor;
visitor.addHandler<std::string>([](const std::string& s) { std::cout << s; });
// ...
Poco::Dynamic::Var v = getVar();
if (visitor.visit(v))
{
    // successfully processed
}
```

### CODE REVIEW

File: Foundation/include/Poco/Dynamic/VarVisitor.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	
+public:
+	template <typename T>
+	bool addHandler(const Details::Handler<T> &f)
+	/// Add handler for specific type T which holds in Var
+	/// This method is more safe, because it saves copy of handler : lambda or std::function
+	/// Returns true if handler was added
+	{
+		auto result = _handlers.emplace(std::ref(typeid(T)),
+										Details::HandlerCaller([handler = f](const Poco::Dynamic::Var& x)
```
Comment (aleks-f) [inline Foundation/include/Poco/Dynamic/VarVisitor.h:71]: too much indent (3 is enough)

File: Foundation/include/Poco/Dynamic/VarVisitor.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	
+	template <typename T>
+	bool addHandler(Details::HandlerPointer<T> f)
+	/// Add handler for specific type T which holds in Var
+	/// This method is less safe, because it saves only copy of function pointer
+	/// Returns true if handler was added
+	{
+		auto result = _handlers.emplace(std::ref(typeid(T)),
+										Details::HandlerCaller([handlerPointer = f](const Poco::Dynamic::Var& x)
```
Comment (aleks-f) [inline Foundation/include/Poco/Dynamic/VarVisitor.h:85]: too much indent (3 is enough)

File: Foundation/src/VarVisitor.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//
+
+#include "Poco/Dynamic/VarVisitor.h"
+
+namespace Poco {
+namespace Dynamic {
+
+bool Visitor::visit(const Var &x) const
+{
+	bool wasHendled = false;
```
Comment (aleks-f) [inline Foundation/src/VarVisitor.cpp:22]: spelling, wasHandled, not hendled

File: Foundation/testsuite/src/VarTest.cpp [hunk 4]
```diff
@@ -3103,6 +3110,169 @@ void VarTest::testSharedPtr()
 	assertTrue(p.referenceCount() == 1);
 }
 
+struct ProcessDummy
+{
+	Var &v;
+	ProcessDummy(Var &var) : v(var) {}
+	void operator()(const Dummy &d) {
```
Comment (aleks-f) [inline Foundation/testsuite/src/VarTest.cpp:3117]: style

**Review by aleks-f — CHANGES_REQUESTED** [review]
@bas524 we need that CI passing. rebase your branch with `devel` to get the most recent changes and I will restart any failing CI (sometimes they just timeout) that does not show a serious bug

File: Foundation/testsuite/src/VarTest.cpp (hunk 4 above)
Comment (bas524) [reply Foundation/testsuite/src/VarTest.cpp]: done

File: Foundation/src/VarVisitor.cpp (hunk 3 above)
Comment (bas524) [reply Foundation/src/VarVisitor.cpp]: done

File: Foundation/include/Poco/Dynamic/VarVisitor.h (hunk 2 above)
Comment (bas524) [reply Foundation/include/Poco/Dynamic/VarVisitor.h]: done

File: Foundation/include/Poco/Dynamic/VarVisitor.h (hunk 1 above)
Comment (bas524) [reply Foundation/include/Poco/Dynamic/VarVisitor.h]: done

**Pull request merged**